### PR TITLE
Fix : solving data fetching for serviceItems

### PR DESCRIPTION
**Pull request by Sherif213**
Solving database document replication

### CODE REVIEW

File: src/COMPONENTS/AboutPage/WhatWeDo.jsx [hunk 1]
```diff
@@ -28,27 +28,30 @@ const WhatWeDo = () => {
...
+
   // Fetch headers from Firestore on mount
   useEffect(() => {
     const fetchHeader = async () => {
       try {
         const headerData = await getWhatFromDB();
         console.log('Fetched about data:', headerData);
-  
+
+        setWhatId(headerData[0].id);
```
Comment (Sherif213) [inline src/COMPONENTS/AboutPage/WhatWeDo.jsx:52]: Here I made the document to be saved for future updates before fetching

File: src/COMPONENTS/AboutPage/WhatWeDo.jsx [hunk 2]
```diff
@@ -28,27 +28,30 @@ const WhatWeDo = () => {
...
+        setSiId(headerData[0].id);
```
Comment (Sherif213) [inline src/COMPONENTS/AboutPage/WhatWeDo.jsx:53]: Here I made the document to be saved for future updates before fetching

File: src/COMPONENTS/AboutPage/WhatWeDo.jsx [hunk 3]
```diff
@@ -60,29 +63,29 @@ const WhatWeDo = () => {
...
-        // Fetch service items if they exist
-        if (headerData[0].serviceItems) {
-          setTypographyContent0({
-            en: headerData[0].en.serviceItems || serviceItems.map(card => ({ label: card.label})),
-            ar: headerData[0].ar.serviceItems || serviceItems.map(card => ({ label: ''})),
-          });
-  
-          setSiId(headerData[0].id);  // Set service items document ID
+          
+          if (headerData[0].en.serviceItems || headerData[0].ar.serviceItems) {
```
Comment (Sherif213) [inline src/COMPONENTS/AboutPage/WhatWeDo.jsx:67]: In here I checked header data in database for both english version and arabic version

File: src/COMPONENTS/AboutPage/WhatWeDo.jsx [hunk 4]
```diff
@@ -136,14 +139,20 @@ const WhatWeDo = () => {
           });
         } else {
           // If serviceItems document does not exist, create a new one
-          const newId = await saveWhatToDB(updatedServiceItemsContent);
-          setSiId(newId);
+          if (siId) {
```
Comment (Sherif213) [inline src/COMPONENTS/AboutPage/WhatWeDo.jsx:142]: Here is a better handling when the document does not exist , it creates new one but if it does exist it doesnot

File: src/COMPONENTS/AboutPage/WhatWeDo.jsx [hunk 5]
```diff
@@ -164,189 +173,195 @@ const WhatWeDo = () => {
           });
         } else {
           // Save a new document if it doesn't exist
-          const newId = await saveWhatToDB(updatedContent);
-          setWhatId(newId);
+          if (whatId) {
```
Comment (Sherif213) [inline src/COMPONENTS/AboutPage/WhatWeDo.jsx:176]: Here is a better handling when the document does not exist , it creates new one but if it does exist it doesnot
but for WhatId

**Review by Sherif213 — COMMENTED** [review]
Review done

**Review by hagerelzamek — APPROVED** [review]
**_

> Perfecto

_**

**Pull request merged**